README.md

```diff
@@ -245,6 +270,19 @@ class Coordinates(private val x: Int, private val y: Int)
 ```
 
 The answer is yes, as the above class still expresses clear intent, without even bothering the reader with the noise of the how.
+
+Would the `sum` work if you had a return type different than `Coordinates` for
+the method? Does the following signature express intent?
+
+```kotlin
+class Coordinates(private val x: Int, private val y: Int) {
+
+  fun sum(addend: Coordinates): Int {
+    // ...
+  }
+
+```
+
```

Comment:
I would say no, but I am unsure about your example. Are you suggesting the `sum` name is misleading as it reminds of arithmetic sum to the reader so they would expect an Int as result? Or are you trying to make a point against primitive obsession?

Reply:
No, my point is that expressing intent has more dimensions than just function naming. A properly named method with a non-proper signature won't be effective as expressing intention.

Reply:
Got it! I agree, I'll mention it or even touch it as a separate section. Thank you